### ISSUE TITLE
Craft 3?

### ISSUE THREAD

**Issue by bossanova808** [elivz/VzUrl-Craft#15]
Hi Eli

Are you planning on updating this for Craft 3?

(I'm just in the planning stages of moving a massive site over to C3 and need to work out what to do witha  few non standard field types like this...)

I presume, if you are not, I can convert the fields back to standard text fields without losing data??

**Comment by elivz** [comment]
Yes, I've got a beta version of the conversion, and am going to try to get it finished up and released soon. I had a baby right around the time Craft3 came out, and have been playing catch-up ever since...

In the mean time, yes: you can freely switch batch and forth between plain-text and url fieldtypes with no data loss.

**Comment by bossanova808** [comment]
That's great Eli & congratulations.  Enjoy that baby and don't hurry with this!

**Comment by bossanova808** [comment]
No pressure, and hope you are enjoying your family! .... but, just wondering if you have a timescale on this at all.  Fine to say 'no' or 'it is miles away' - just so I know & can plan.

**Comment by bossanova808** [comment]
Hi Eli

This is great!

One question, I've noticed a bunch of field types have not included migrations for exisiting fields for C2 -> C3 upgrades, so I have had to write my own.  

Writing my own is fine of course, but it's handy for everyone if the plugin has this built in.  I couldn't see anything in a quick look at your code to do this...maybe you could add it.

Here's some example code you could adapt for that should make it very easy...you'd run this on install, I suppose (I run it as part of a custom console controller after I do my main db upgrades).

```
        // LJ DYNAMIC FIELDS

        // Look for any old LJ Checkbox fields and in this case we also need to update their settings
        $fields = (new \craft\db\Query())
            ->select(['id', 'settings'])
            ->from(['{{%fields}}'])
            ->where(['in', 'type', ['Lj_DynamicFields_Checkboxes']])
            ->all();

        if($fields)
        {
            // Update each field and its settings
            foreach($fields as $field)
            {
                $oldSettings = $field['settings'] ? Json::decode($field['settings']) : null;
                $newSettings = $this->_migrateLJCheckboxSettings($oldSettings);

                $db->createCommand()->update('{{%fields}}', [
                    'type' => CheckboxesField::class,
                    'settings' => Json::encode($newSettings)
                ], ['id' => $field['id']])->execute();
            }
        }
 ```

Hope you & your little one are well!

**Comment by elivz** [comment]
I added that migration for the field settings. Thanks for the tip!
